App.js

```diff
@@ -1,69 +1,182 @@
-
-        switch(routeName) {
-          case "Home":
-            icon = faHome
-            break;
-          case "Schedule":
-            icon = faList
-            break;
-          case "Notifications":
-            icon = faBell 
-            break;
-          default:
-            icon = null;
-        }
+    Workshops
+});
 
-        return <FontAwesomeIcon color={tintColor} icon={icon} />
-    }
+const LoginStack = createStackNavigator({
+    Login
+});
+
+const TabNavigator = createBottomTabNavigator(
+    {
+        Home: HomeStack,
+        Schedule: ScheduleStack,
+        Notifications: WorkshopStack,
+        Login: LoginStack
+    },
+    {
+        defaultNavigationOptions: ({ navigation }) => ({
+            tabBarIcon: ({ focused, horizontal, tintColor }) => {
+                const { routeName } = navigation.state;
+
+                switch (routeName) {
+                    case "Home":
+                        icon = faHome;
+                        break;
+                    case "Schedule":
+                        icon = faList;
+                        break;
+                    case "Notifications":
+                        icon = faBell;
+                        break;
+                    case "Login":
+                        icon = faKey;
+                        break;
+                    default:
+                        icon = faQuestion;
+                }
 
-    })
-  });
+                return <FontAwesomeIcon color={tintColor} icon={icon} />;
+            }
+        })
+    }
+);
 
 const AppContainer = createAppContainer(TabNavigator);
 
 export default class App extends Component<Props> {
+    state = {
+        user: null,
+        accessToken: null
+    };
+
+    componentDidMount() {
+        AsyncStorage.getItem(
+            "userData",
+            (error, result) =>
+                result && this.setState({ user: JSON.parse(result) })
+        );
+        AsyncStorage.getItem(
+            "accessToken",
+            (error, result) => result && this.setState({ accessToken: result })
+        );
+    }
 
-  render() {
-    return (
-      <AppContainer>Hi</AppContainer>
-    )
-  }
-}
\ No newline at end of file
+    logout = async () => {
+        fetch(`${authUrl}/api/user/logout`, {
+            method: "POST",
+            headers: {
+                Authorization: "Bearer " + this.state.accessToken
+            }
+        }).then(response => {
+            AsyncStorage.removeItem("accessToken");
+            AsyncStorage.removeItem("userData");
+            this.setState({ accessToken: "", user: null });
+        });
+    };
+
+    login = async () => {
+        try {
+            const result = await authorize(config);
+
+            this.setState({ accessToken: result.accessToken });
+
+            this.fetchUserDetails(result.accessToken);
+
+            AsyncStorage.setItem("accessToken", accessToken);
+        } catch (error) {
+            console.log(error);
```

Comment:
Open an issue and log a todo comment about surfacing errors to UI

Reply:
I think also mention in the issue - surface login success greeting/land user on some page that indicates login success